src/components/useFetch.ts

```diff
@@ -0,0 +1,38 @@
+import { useEffect, useState } from "react";
+
+interface FetchState<T> {
+  data: T | null;
+  isLoading: boolean;
+  error: string | null;
+}
+
+export const useFetch = <T>(url: string | string[]): FetchState<T[]> => {
+  const [data, setData] = useState<T[] | null>(null);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [error, setError] = useState<string | null>(null);
+
+  useEffect(() => {
+    const fetchSingleUrl = async (singleUrl: string): Promise<T> => {
```

Comment:
```suggestion
    const handleFetchData = async (singleUrl: string): Promise<T> => {
```
Single тут вообще лишнее слово, из названия итак должно быть понятно сколько сущностей запрашивается:
fetchUser - 1 user
fetchUsers - много юзеров
А fetch-ишь ты не урл, а какую-то бизнесовую сущность